models/servers.go

```diff
@@ -0,0 +1,100 @@
+package models
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// Server maps AsyncAPI serves.item object
+type Server struct {
+	Extensions    Extensions       `json:"-"`
+	URL           string           `json:"url"`
+	Description   string           `json:"description,omitempty"`
+	Scheme        string           `json:"scheme"`
+	SchemeVersion string           `json:"schemeVersion,omitempty"`
+	Variables     *ServerVariables `json:"variables,omitempty"`
+	BaseChannel   string           `json:"baseChannel,omitempty"`
+	// Security      []*SecurityRequirement `json:"security,omitempty"`
+}
+
+// Unmarshal unmarshals JSON
+func (value *Server) Unmarshal(data []byte) error {
+	return json.Unmarshal(data, value)
+}
+
+// Marshal marshals JSON
+func (value *Server) Marshal() ([]byte, error) {
+	if value.Extensions == nil {
+		return json.Marshal(value)
+	}
+	return MarshalWithExtensions(value, value.Extensions)
+}
+
+// ServerVariables object
+type ServerVariables struct {
+	AdditionalProperties map[string]*ServerVariable `json:",omitempty"`
+}
```

Comment:
You can avoid having this weird "AdditionalProperties" by doing the following:

```go
type ServerVariables map[string]*ServerVariable
```

Then you reference it above:

```go
Variables     *ServerVariables `json:"variables,omitempty"`
```

See this example: https://github.com/asyncapi/parser/blob/master/models/extension.go#L10

Reply:
great, I knew there had to be a better way :)